lib/datasource/local/flutter_js/web_view_runner.dart

```diff
@@ -101,9 +106,15 @@ class WebViewRunner {
               jsCodeStarted = 0;
             }
           }
-          if (message.message.contains("WebSocket is not connected") && socketDisconnectedAction != null) {
-            socketDisconnectedAction();
+          if (message.message.contains("API-WS: disconnected from") && this.socketDisconnectedAction != null) {
```

Comment:
String matching : these checks seems very brittle. I am not sure if there is any other option for this.

Reply:
Yes - it's totally unreliable.

I actually disabled the function I was calling for it

Websocked is not connected gets called whenever there's a tiny blip in connectivity, so doesn't mean much

I got API-WS: disconnected from... when there was a non-recoverable error, e.g. laptop was sleeping for significant time, and no way to reconnect. 

Problem is we don't get good feedback from JavaScript. 

So for now we have our external flutter based pulse checker, which periodically makes an API call. 

When that fails, I restart everything. Which works OK I think, but this is still a draft PR, need to do more testing.